lib/Yield/Assembly/HTML.cpp

```diff
@@ -0,0 +1,334 @@
+/// \file HTML.cpp
+/// \brief
+
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+#include "llvm/Support/FormatVariadic.h"
+
+#include "revng/EarlyFunctionAnalysis/FunctionMetadata.h"
+#include "revng/Model/Binary.h"
+#include "revng/Yield/Assembly/Assembly.h"
+#include "revng/Yield/HTML.h"
+
+namespace yield::html {
+
+namespace tags {
+static constexpr auto NormalBasicBlockLabel = "basic-block-label";
+static constexpr auto HiddenBasicBlockLabel = "hidden-basic-block-label";
+static constexpr auto GraphOnlyBasicBlockLabel = "graph-only-basic-block-label";
+
+static constexpr auto Instruction = "instruction";
+static constexpr auto InstructionAddress = "instruction-address";
+static constexpr auto InstructionBytes = "instruction-bytes";
+
+static constexpr auto InstructionMnemonic = "mnemonic";
+static constexpr auto InstructionMnemonicPrefix = "mnemonic-prefix";
+static constexpr auto InstructionMnemonicSuffix = "mnemonic-suffix";
+static constexpr auto InstructionOpcode = "instruction-opcode";
+
+static constexpr auto Comment = "comment";
+static constexpr auto Error = "error";
+static constexpr auto ImmediateValue = "immediate-value";
+static constexpr auto MemoryOperand = "memory-operand";
+static constexpr auto Register = "register";
+
+static constexpr auto InstructionTarget = "instruction-target";
+static constexpr auto InstructionTargets = "instruction-targets";
+
+static constexpr auto ExternalWhitespace = "external-whitespace";
+static constexpr auto InternalWhitespace = "internal-whitespace";
+static constexpr auto Untagged = "untagged";
+} // namespace tags
+
+namespace templates {
+static constexpr auto Span = "<span class='{0}'>{1}</span>";
+static constexpr auto Div = "<div class='{0}'>{1}</div>\n";
```

Comment:
* I'd prefer `"` as delimiters. `'` seem quite 90s.
* About the following:
  ```html
  <span class='instruction-address'>0x40055c:Code_x86_64</span>
  ```
  I'd like this to become
  ```html
  <div class='instruction' id="block_0x40055c:Code_x86_64">
  ```
  If we make the link become `<a href="#block_0x40055c:Code_x86_64">`, we get intra-function navigation for free!
* Let's not print `<a class='unknown-link' href='revng://unknown'>(unknown)</a>`, no point.

Reply:
I'd also like to see the list of targets after a `;` (even if probably the comment character varies on an asm-by-asm basis). Can you update the CSS?

Reply:
Sure.

Reply:
I looked into that, it shouldn't be a problem to request what the comment character is for a given architecture from llvm, so that's going to be fine as well.

Reply:
Enterprise

Reply:
1 is also done: https://github.com/revng/revng/pull/173/commits/8350ef4214d5b1cae6cad28ffb0166b95d56ab81.